tests/test_framework/phpbb_functional_test_case.php

```diff
@@ -410,7 +410,26 @@ public function install_ext($extension)
 		$form = $crawler->selectButton('Enable')->form();
 		$crawler = self::submit($form);
 		$this->add_lang('acp/extensions');
-		$this->assertContainsLang('EXTENSION_ENABLE_SUCCESS', $crawler->filter('div.successbox')->text());
+
+		$meta_refresh = $crawler->filter('meta[http-equiv="refresh"]');
+
+		// Wait for extension to be fully enabled
+		while (sizeof($meta_refresh))
+		{
+			preg_match('#url=.+/(adm+.+)#', $meta_refresh->attr('content'), $match);
+			$url = $match[1];
+			$crawler = self::request('POST', $url);
+			$meta_refresh = $crawler->filter('meta[http-equiv="refresh"]');
+		}
+
+		if (!empty($meta_refresh))
+		{
+			$this->assertContainsLang('EXTENSIONS_ADMIN', $crawler->filter('.main > h1')->text());
+		}
+		else
+		{
+			$this->assertContainsLang('EXTENSION_ENABLE_SUCCESS', $crawler->filter('div.successbox')->text());
```

Comment:
shouldn't there always be a successbox at the end, because you loop, until `!empty($meta_refresh)` is true?